src/bun.js/bindings/OnigurumaRegExp.cpp

```diff
@@ -95,6 +98,58 @@ static WTF::String extendMultibyteHexCharacters(const WTF::String& string)
             break;
         }
 
+        // convert character properties
+        if (characters[i] == '{' && i - 1 >= 0 && (characters[i - 1] == 'p' || characters[i - 1] == 'P') && i - 2 >= 0 && characters[i - 2] == '\\') {
```

Comment:
we only need to check for `i - 2 >= 0` right instead of i -1 >= 0, right?

also, since this will be uncommon let's do `UNLIKELY`

```suggestion
       // \\p{
       // \\P{
        if (UNLIKELY(characters[i] == '{' && i >= 2 && (characters[i - 1] == 'p' || characters[i - 1] == 'P') && characters[i - 2] == '\\')) {
```